packages/docs/docs/advanced/custom-components.mdx

```diff
@@ -0,0 +1,141 @@
+`Node`. It is advisable to extend from a component most similar to the custom component you are building. For instance, 
+if you were to make a component including a [Layout](/api/2d/components/Layout), you should extend [Layout](/api/2d/components/Layout)
+and [LayoutProps](/api/2d/components/LayoutProps).
+```tsx
+export interface SwitchProps extends NodeProps { ... }
+
+export class Switch extends Node { ... }
+```
+
+To use the properties defined in the interface, your class _must_ contain a property with the
+same name. Motion Canvas provides type decorators to facilitate this like `@initial()` and
+`@signal()`. Click [here](/docs/signals) for more information on signals.
+
+Here is an example of how you would define such properties:
+```tsx
+export class Switch extends Node {
+    // @initial - sets the property to an
+    // initial value if it was not provided.
+    // The @initial decorator is optional.
+    @initial(false)
+    // @signal - is required by motion canvas
+    // for every prop that was passed in.
+    @signal()
+    public declare readonly initialState: SimpleSignal<boolean, this>
+
+    @initial("#68ABDF")
+    @signal()
+    public declare readonly accent: ColorSignal<this>
+
+    ...
+
+}
+```
+Notice how colors are wrapped in `ColorSignal<>` whilst any other type (even user-defined ones)
+are wrapped in `SimpleSignal<>`. The type does not need to be passed to color signal as Motion
+Canvas knows that it must be of a color-resolvable type. In both, the class is passed at the end
+of the wrapper to register the signal to the class. The properties must also be _declared_, _public_ 
+and _readonly_.
+
+Normal properties of a component class can be defined as normal. For example like so:
+```tsx
+private isOn: boolean;
+```
+
+Every component _requires_ a constructor. The format of the constructor will always
+look similar to this, with optional argument for the properties:
+```tsx
+public constructor(props?: SwitchProps) {
+    super({
+        // If you wanted to ensure that layout was always
+        // true for this component, you could add it here
+        // as such:
+
+        //layout: true
+        ...props
+    })
+    ...
+}
+```
+The `props` parameter can also be useful outside of the `super()` call to access your data elsewhere. 
+For example, if you were building a component to display an array, you could use props to set the color
+of every [Rect](/api/2d/components/Rect) in the array.
+
+Now we can add elements to the view by using `this.add()`, much like you would add to a
+scene's view:
+```tsx
+public constructor(props?: SwitchProps) {
+
+    ...
+
+    this.add(
+            <Rect
+                ...
+            >
+                <Circle 
+                    ... 
+                />
+            </Rect>
+    )
+}
+```
+Since this is also a regular typescript class, you can add methods. This is especially useful
```

Comment:
Did you decide not to use "also" here?

Reply:
Since this is still... was transformed into Since this is also. is that not what you meant?

Reply:
I was intending "Since this is a regular typescript class, you can also add methods."

Reply:
should typescript be omitted since its implied that typescript is used throughout the project?

Reply:
I think typescript is a good distinction because we're talking about component classes as well, so having a distinction between a "regular TypeScript class" and the rules for a component class is good. We might capitalize TypeScript, though.